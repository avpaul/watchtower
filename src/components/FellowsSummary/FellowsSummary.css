.ops-dashboard__fellows-summary {
  padding-left: 4.166666666%;
  padding-right: 4.166666666%;
}

.ops-dashboard__filter {
  justify-content: center;
  margin-bottom: 28px;
  padding-right: 4.166666%;
}

.ops-dashboard__filter .contain {
  padding-left: 1%;
  padding-right: 0;
}

.ops-dashboard__fellow-summary-text {
  color: #0459e4;
  font-family: 'DIN Pro';
  font-size: 1.25rem;
  font-weight: bold;
  line-height: 1.625rem;
  margin-top: 3.75rem;
}

.ops-dashboard__filter .number {
  padding: 0 2rem 1.375rem 0;
  width: 100%;
  margin: 0;
}

.ops-dashboard__filter .card {
  margin: 20px auto 0 auto;
}

.total-fellows-card {
  font-weight: bolder;
}

@media screen and (max-width: 700px) {
  .ops-dashboard__filter {
    justify-content: left;
  }

  .ops-dashboard__filter .contain {
    padding-left: 4%;
    padding-right: 0;
  }
}

@media screen and (min-width: 729px) {
  .ops-dashboard__filter {
    justify-content: left;
  }

  .ops-dashboard .row {
    margin-left: 0;
  }

  .ops-dashboard__fellow-summary-text {
    text-align: left;
  }

  .ops-dashboard__filter .card {
    border: none;
    margin: 20px 20px 28px 0;
  }
}

.scroll-wrapper {
  overflow-x: hidden;
  overflow-y: hidden;
  white-space: nowrap;
}

.card-scroll-wrapper {
  display: inline-block;
}
