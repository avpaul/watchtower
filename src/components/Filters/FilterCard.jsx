import React from 'react';
import PropTypes from 'prop-types';
import './Filters.css';
/**
 *
 * FilterCard Component
 * @param {Object} props
 *
 * @returns {JSX} React JSX
 */
const FilterCard = ({ className, onClick, filterId, cardDetails }) => (
  <div
    className={className}
    id={filterId}
    onClick={onClick}
    onKeyPress={onClick}
    role="button"
    tabIndex="0"
  >
    <p className="title">{cardDetails.title}</p>
    <p className="sub">{cardDetails.subTitle}</p>
    <p className="number">{cardDetails.totalFellows}</p>
  </div>
);

FilterCard.defaultProps = {
  className: 'filterCard'
};

FilterCard.propTypes = {
  className: PropTypes.string,
  cardDetails: PropTypes.shape({
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string,
    totalFellows: PropTypes.oneOfType([PropTypes.number, PropTypes.string])
      .isRequired
  }).isRequired,
  filterId: PropTypes.oneOfType([PropTypes.number, PropTypes.string])
    .isRequired,
  onClick: PropTypes.func.isRequired
};

export default FilterCard;
