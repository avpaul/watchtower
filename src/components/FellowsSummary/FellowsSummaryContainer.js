import { connect } from 'react-redux';

import FellowsSummary from './FellowsSummary';

export const mapStateToProps = state => ({
  fellowsSummary: state.opsDashboard.fellowsSummary
});

export default connect(mapStateToProps)(FellowsSummary);
