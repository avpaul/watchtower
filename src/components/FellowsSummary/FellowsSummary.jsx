import React from 'react';
import PropTypes from 'prop-types';
import Slider from 'react-slick';
import { carouselOptions } from '../../utils';
import FilterCard from '../Filters/FilterCard';
import FellowSummaryLabel from '../FellowSummaryLabel';
import './FellowsSummary.css';

const formatCards = fellowLevels => {
  const keys = fellowLevels ? Object.keys(fellowLevels) : '';
  const result = keys.map(key => ({
    title: key,
    totalFellows: parseInt(fellowLevels[key], 10)
  }));
  return result;
};

const FellowsSummary = props => {
  const {
    fellowsSummary: {
      fellowsSummaryToday: { latestWeekSummary = {} }
    },
    handleCardClick,
    displayByRole
  } = props;

  const fellowsSummaryCard = formatCards(latestWeekSummary);
  return (
    <div className="ops-dashboard__fellows-summary">
      <FellowSummaryLabel />
      <div className="row ops-dashboard__filter">
        <Slider {...carouselOptions(3)}>
          {fellowsSummaryCard
            .filter(card => {
              if (!displayByRole) return true;
              switch (true) {
                case !!displayByRole.WATCH_TOWER_EM:
                  return card.title === 'D0B';
                case !!displayByRole.WATCH_TOWER_SL:
                  return card.title === 'D0A';
                default:
                  return true;
              }
            })
            .map(fellowSummary => (
              <div className="p-1">
                <FilterCard
                  key={fellowSummary.title}
                  filterId={fellowSummary.title}
                  cardDetails={fellowSummary}
                  className="card"
                  onClick={handleCardClick}
                />
              </div>
            ))}
        </Slider>
      </div>
    </div>
  );
};

FellowsSummary.propTypes = {
  displayByRole: PropTypes.objectOf(
    PropTypes.oneOfType([PropTypes.string, PropTypes.number])
  ).isRequired,
  fellowsSummary: PropTypes.instanceOf(Object).isRequired,
  handleCardClick: PropTypes.func.isRequired
};

export default FellowsSummary;
