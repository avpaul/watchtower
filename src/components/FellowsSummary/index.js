import FellowsSummaryContainer from './FellowsSummaryContainer';

export default FellowsSummaryContainer;
