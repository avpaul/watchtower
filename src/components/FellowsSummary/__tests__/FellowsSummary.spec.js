import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import FellowsSummary from '../FellowsSummary';
import FilterCard from '../../Filters/FilterCard';

const setup = propOverrides => {
  const props = {
    fellowsSummary: [
      { id: 'testkey1', title: 'Test key 1', totalFellows: 10 },
      { id: 'testkey2', title: 'Test key 2', totalFellows: 5 },
      { id: 'testkey3', title: 'Test key 3', totalFellows: 5 }
    ],
    handleCardClick: jest.fn(),
    ...propOverrides
  };

  const wrapper = shallow(<FellowsSummary {...props} />);

  return {
    props,
    wrapper,
    count: wrapper.find(FilterCard).length
  };
};

describe('<FellowsSummary />', () => {
  it('renders without crashing', () => {
    const { props } = setup();
    const div = document.createElement('div');
    ReactDOM.render(<FellowsSummary {...props} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders three(3) <FilterCard />', () => {
    const { count } = setup();
    expect(count).toEqual(3);
  });

  it('handles click events on the <FilterCard />', () => {
    const {
      props: { handleCardClick },
      wrapper
    } = setup();
    wrapper
      .find(FilterCard)
      .first()
      .simulate('click');
    expect(handleCardClick).toHaveBeenCalledTimes(1);
  });
});
