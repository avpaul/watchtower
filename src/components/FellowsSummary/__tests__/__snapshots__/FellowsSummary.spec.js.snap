// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FellowsSummary /> renders to match snapshots 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FellowsSummary
    displayByRole={
        Object {
            "WATCH_TOWER_OPS": "343242434",
          }
    }
    fellowsSummary={
        Object {
            "D0A": 5,
            "D0B": 5,
            "Total": 10,
          }
    }
    handleCardClick={[Function]}
    loading={false}
    loggedInRole="WATCH_TOWER_OPS"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Title
          subTitle=""
          title="FELLOWS SUMMARY"
/>,
        <div
          className="row ops-dashboard__filter"
>
          <FiltersView
                    displayByRole={
                              Object {
                                        "WATCH_TOWER_OPS": "343242434",
                                      }
                    }
                    filterCardClassName="p-1"
                    filters={
                              Array [
                                        Object {
                                          "heading": "Total D0 Fellows",
                                          "subTitle": "Click to see details",
                                          "title": "Total",
                                          "totalFellows": 10,
                                        },
                                        Object {
                                          "heading": "Total D0A Fellows",
                                          "subTitle": "Click to see details",
                                          "title": "D0A",
                                          "totalFellows": 5,
                                        },
                                        Object {
                                          "heading": "Total D0B Fellows",
                                          "subTitle": "Click to see details",
                                          "title": "D0B",
                                          "totalFellows": 5,
                                        },
                                      ]
                    }
                    handleCardClick={[Function]}
          />
</div>,
      ],
      "className": "ops-dashboard__fellows-summary",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "subTitle": "",
          "title": "FELLOWS SUMMARY",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <FiltersView
            displayByRole={
                        Object {
                                    "WATCH_TOWER_OPS": "343242434",
                                  }
            }
            filterCardClassName="p-1"
            filters={
                        Array [
                                    Object {
                                      "heading": "Total D0 Fellows",
                                      "subTitle": "Click to see details",
                                      "title": "Total",
                                      "totalFellows": 10,
                                    },
                                    Object {
                                      "heading": "Total D0A Fellows",
                                      "subTitle": "Click to see details",
                                      "title": "D0A",
                                      "totalFellows": 5,
                                    },
                                    Object {
                                      "heading": "Total D0B Fellows",
                                      "subTitle": "Click to see details",
                                      "title": "D0B",
                                      "totalFellows": 5,
                                    },
                                  ]
            }
            handleCardClick={[Function]}
/>,
          "className": "row ops-dashboard__filter",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "displayByRole": Object {
              "WATCH_TOWER_OPS": "343242434",
            },
            "filterCardClassName": "p-1",
            "filters": Array [
              Object {
                "heading": "Total D0 Fellows",
                "subTitle": "Click to see details",
                "title": "Total",
                "totalFellows": 10,
              },
              Object {
                "heading": "Total D0A Fellows",
                "subTitle": "Click to see details",
                "title": "D0A",
                "totalFellows": 5,
              },
              Object {
                "heading": "Total D0B Fellows",
                "subTitle": "Click to see details",
                "title": "D0B",
                "totalFellows": 5,
              },
            ],
            "handleCardClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Title
            subTitle=""
            title="FELLOWS SUMMARY"
/>,
          <div
            className="row ops-dashboard__filter"
>
            <FiltersView
                        displayByRole={
                                    Object {
                                                "WATCH_TOWER_OPS": "343242434",
                                              }
                        }
                        filterCardClassName="p-1"
                        filters={
                                    Array [
                                                Object {
                                                  "heading": "Total D0 Fellows",
                                                  "subTitle": "Click to see details",
                                                  "title": "Total",
                                                  "totalFellows": 10,
                                                },
                                                Object {
                                                  "heading": "Total D0A Fellows",
                                                  "subTitle": "Click to see details",
                                                  "title": "D0A",
                                                  "totalFellows": 5,
                                                },
                                                Object {
                                                  "heading": "Total D0B Fellows",
                                                  "subTitle": "Click to see details",
                                                  "title": "D0B",
                                                  "totalFellows": 5,
                                                },
                                              ]
                        }
                        handleCardClick={[Function]}
            />
</div>,
        ],
        "className": "ops-dashboard__fellows-summary",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "subTitle": "",
            "title": "FELLOWS SUMMARY",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <FiltersView
              displayByRole={
                            Object {
                                          "WATCH_TOWER_OPS": "343242434",
                                        }
              }
              filterCardClassName="p-1"
              filters={
                            Array [
                                          Object {
                                            "heading": "Total D0 Fellows",
                                            "subTitle": "Click to see details",
                                            "title": "Total",
                                            "totalFellows": 10,
                                          },
                                          Object {
                                            "heading": "Total D0A Fellows",
                                            "subTitle": "Click to see details",
                                            "title": "D0A",
                                            "totalFellows": 5,
                                          },
                                          Object {
                                            "heading": "Total D0B Fellows",
                                            "subTitle": "Click to see details",
                                            "title": "D0B",
                                            "totalFellows": 5,
                                          },
                                        ]
              }
              handleCardClick={[Function]}
/>,
            "className": "row ops-dashboard__filter",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "displayByRole": Object {
                "WATCH_TOWER_OPS": "343242434",
              },
              "filterCardClassName": "p-1",
              "filters": Array [
                Object {
                  "heading": "Total D0 Fellows",
                  "subTitle": "Click to see details",
                  "title": "Total",
                  "totalFellows": 10,
                },
                Object {
                  "heading": "Total D0A Fellows",
                  "subTitle": "Click to see details",
                  "title": "D0A",
                  "totalFellows": 5,
                },
                Object {
                  "heading": "Total D0B Fellows",
                  "subTitle": "Click to see details",
                  "title": "D0B",
                  "totalFellows": 5,
                },
              ],
              "handleCardClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
